### ISSUE TITLE
Empty http response when both AddAuditFilter and UseAuditMiddleware

### ISSUE THREAD

**Issue by henkosch** [thepirat000/Audit.NET#161]
**Describe the bug**
I want to use both the asp.net core action filter and the middleware to generate audit events. If I understand correctly the middleware can be a fallback for requests fail to enter the mvc action filter pipeline. I want to include the response body to the audit event but I receive an empty http response in the http client.

```c#
services.AddMvc(mvc =>
{
    mvc.AddAuditFilter(config => config
        .LogAllActions()
        .WithEventType("action")
        .IncludeHeaders()
        .IncludeRequestBody()
        .IncludeModelState()
        .IncludeResponseBody());
});
```

```c#
app.UseAuditMiddleware(config => config
   .WithEventType("request")
   .IncludeHeaders()
   .IncludeRequestBody()
   .IncludeResponseBody() // !!!! BUG: this is causing the final http response to be empty
);
```

**To Reproduce**
I have created a sample project to reproduce the problem:
https://github.com/henkosch/auditnet-elasticsearch-bug-sample

You can fix the issue by commenting out this line:
https://github.com/henkosch/auditnet-elasticsearch-bug-sample/blob/master/Startup.cs#L117

To start elasticsearch and kibana for the project use the docker-compose.yml in the root directory:
docker-compose up

After starting the project you can use the swagger ui to call the api:
http://localhost:5000/swagger

**Expected behavior**
The audit event is written into the data store with the response body included and also the http client receives the response body.

**Libraries (specify the Audit.NET extensions being used including version):**
For example:
 - Audit.NET: 13.1.1
 - Audit.NET.Elasticsearch: 13.1.1
 - Audit.WebApi: 13.1.1

**Comment by henkosch** [comment]
See #159 for previous discussion.

**Comment by thepirat000** [comment]
I see where the bug is... there is a missing Seek on the response stream, let me add some tests and fix this

**Comment by thepirat000** [comment]
This was fixed for version 13.1.2, please upgrade and reply to this if you still have problems.
